pallets/pallet-bonded-coins/src/lib.rs

```diff
@@ -363,8 +363,15 @@ pub mod pallet {
 
 			let currency_length = currencies.len();
 
-			let currency_ids: BoundedVec<FungiblesAssetIdOf<T>, T::MaxCurrenciesPerPool> =
-				T::NextAssetIds::try_get(currency_length.saturated_into()).map_err(|err| err.into())?;
+			let currency_ids = T::NextAssetIds::try_get(currency_length.saturated_into())
+				.map_err(|_| Error::<T>::Internal)
```

Comment:
Still very much opposed to the overuse of non-informative "Internal" errors. Why mask hook errors instead of showing what is actually going on? If in the hook I throw an `Arithmetic::Overflow` because the space of ids is exhausted, I would expect this to be seen by the caller instead of a non-informative "well, that wasn't supposed to happen".